include/mesh.tcc

```diff
@@ -940,8 +940,11 @@ std::vector<Eigen::Matrix<double, 3, 1>> mpm::Mesh<Tdim>::particles_vector_data(
       Eigen::Vector3d data;
       data.setZero();
       auto pdata = (*pitr)->vector_data(attribute);
-      // Fill stresses to the size of dimensions
-      for (unsigned i = 0; i < Tdim; ++i) data(i) = pdata(i);
+      // Fill data attributes to size of 3
+      const unsigned attribute_size =
```

Comment:
I'm sorry, I don't understand what's happening here, why all these std::min and max?

Reply:
`vtk` has to be a component of 3, so we cannot iterate more than 3.
In cases of `displacement` and `velocities` in 2D, we want to only output 2 components, not 3 since `pdata` is a vector of dimension 2.
But in cases of `stresses` in 2D, we want to output all 3 components, not only 2. So in this case `pdata` is a vector of dimension 6, so we take the first 3.

Therefore we need to do this. Maybe you have a better idea?

Reply:
@ezrayst I think the prior one is okay as @kks32 mentioned. `TDim` will always be 2 or 3 right? So even if `pdata` is 6-component vector, it will only copy the first `TDim` values.

Reply:
But I want to get stress in 3 directions even in 2D

Reply:
The magnitude output of VTK will be incorrect if you copy uncomputed 3rd component of stress in 2D

Reply:
This change affects not just stresses, but other dimensional vectors such as velocities, so I don't think it's a good idea to change this.

Reply:
But the 3rd component of stress in 2D is computed!

Reply:
Zero strain in that direction does not mean zero stress in that direction.